[docs] Improve textfield docs and re-establish links to Material UI components (#3929)
mui/mui-toolpad: docs/data/toolpad/studio/reference/components/autocomplete.md

| <span class="prop-name">defaultValue</span> | <span class="prop-type">string</span>  | <span class="prop-default">""</span>        | A default value.                                                                                                                                                                                                                                                                     |
| <span class="prop-name">label</span>        | <span class="prop-type">string</span>  | <span class="prop-default">"Search…"</span> | The label to display for the autocomplete.                                                                                                                                                                                                                                           |

> | <span class="prop-name">fullWidth</span>    | <span class="prop-type">boolean</span> |                                             | If true, the autocomplete will take up the full width of its container.                                                                                                                                                                                                              |

Reviewer: ```suggestion
| <span class="prop-name">fullWidth</span>    | <span class="prop-type">boolean</span> |                                             | If true, the autocomplete takes up the full width of its container.                                                                                                                                                                                                              |
```

Reviewer: Oh wait this isn't the right place to change it, ignore it, I'll add a commit.

Author: Yeah, I assumed this had to be done somehow automatically.

| <span class="prop-name">size</span>         | <span class="prop-type">string</span>  | <span class="prop-default">"small"</span>   | The size of the autocomplete. One of `small`, `medium`, or `large`.                                                                                                                                                                                                                  |

> | <span class="prop-name">loading</span>      | <span class="prop-type">boolean</span> |                                             | If true, the autocomplete will display a loading indicator.                                                                                                                                                                                                                          |

Reviewer: ```suggestion
| <span class="prop-name">loading</span>      | <span class="prop-type">boolean</span> |                                             | If true, the autocomplete displays a loading indicator.                                                                                                                                                                                                                          |
```

> | <span class="prop-name">disabled</span>     | <span class="prop-type">boolean</span> |                                             | If true, the autocomplete will be disabled.                                                                                                                                                                                                                                          |

Reviewer: ```suggestion
| <span class="prop-name">disabled</span>     | <span class="prop-type">boolean</span> |                                             | If true, the autocomplete is disabled.                                                                                                                                                                                                                                          |
```
